### PR TITLE
Make download_dsyms work for appletvos

### PR DESCRIPTION
**Pull request by tmm1**
I added a `DOWNLOAD_DSYMS_PLATFORM` option, but maybe it's better to use `FASTLANE_PLATFORM_NAME` or the `DEFAULT_PLATFORM` instead?

cc @KrauseFx

### CODE REVIEW

File: spaceship/lib/spaceship/tunes/application.rb [hunk 1]
```diff
@@ -201,8 +201,8 @@ def build_trains
 
       # The numbers of all build trains that were uploaded
       # @return [Array] An array of train version numbers
-      def all_build_train_numbers
-        client.all_build_trains(app_id: self.apple_id).fetch("trains").collect do |current|
+      def all_build_train_numbers(platform: nil)
```
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/application.rb]: Adding the new parameter will not work with accessing this from `download_dsyms`, this have to be 2 separate PRs with a dependency update. So first step would be to split the PR up to be 2 PRs 👍

Comment (tmm1) [inline spaceship/lib/spaceship/tunes/application.rb]: I understand that both gems will need to be released, and the dependency will need to be bumped. But why does it require two PRs?

Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/application.rb]: @tmm1 Good question, that's just how we handled it until now to always be in a consistent state, but I guess you're right

**Comment by KrauseFx** [comment]
> I added a DOWNLOAD_DSYMS_PLATFORM option, but maybe it's better to use FASTLANE_PLATFORM_NAME or the DEFAULT_PLATFORM instead?

I agree this is a great idea. You have more experience with working on Apple TV apps: Is that something you would/could specify in the Appfile? Or maybe have a new platform in the `Fastfile`?

```ruby
platform :ios do

end

platform :tv do

end
```

What do you think?
